modules/app.xqm

```diff
@@ -50,10 +50,37 @@ function app:base-url($node as node(), $model as map(*), $base-url as xs:string)
 }; 
 
 declare
-function app:search-package ($node as node(), $model as map(*), $q as xs:string?) as element(li)* {
-    if (empty($q)) then () else
-    for $package-group in packages:search($q) 
-    return packages:render-list-item($package-group, $node)
+    %templates:replace
+function app:search-package ($node as node(), $model as map(*), $q as xs:string?) as element() {
+    if (empty($q) or string-length($q) eq 0) then (
+        <p>Enter a search term in the input above.</p>
+    ) else (
+        let $results := packages:search($q)
+        
+        return
+            if (count($results) = 0) then (
+                <p>The search for <strong>{$q}</strong> yielded no results.</p>
+            ) else (
+                <ul class="package-list">{
+                    for $package-group in $results
+                    return packages:render-list-item($package-group, <li />)
+                }</ul>
+            )
+    )
+};
+
+declare
+    %templates:replace
+function app:show-top-nav-search ($node as node(), $model as map(*)) as element()? {
+    (: if ($model?show-top-nav-search) then ( :)
+    if ($model?('show-top-nav-search')) then (
+        $node
+    ) else (
+        <span>{
+            $model?('show-top-nav-search'),
+            request:get-parameter("top-nav-search", "no")
+        }</span>
+    )
```

Comment:
I see how the form is displayed if the parameter is present, but if it's not present, why return anything? I'd have expected an empty sequence. (Perhaps the contents of the `<span>` were there for debugging, and now can be removed?)